### ISSUE TITLE
'groupId' not always pulling from parent pom

### ISSUE THREAD

**Issue by roytruelove** [mojohaus/versions-maven-plugin#5]
(with version 2.2)

Getting an NPE with the following (pretty standard) scenario using `versions:set`

In my remote repo (e.g. Artifactory) I've got a global parent pom (com.myco:global-parent-pom:pom.1.0-SNAPSHOT)

I then have a project:

```
+---myProject
    +--- pom.xml
    +--- moduleA
         +--- pom.xml
         +--- (etc)
    +--- moduleB
         +--- pom.xml
         +--- (etc)
```

myProject's parent pom references the 'global' parent pom as its parent:

```
<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
        xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">

        <artifactId>my-project-pom</artifactId>
        <version>1.0.0-SNAPSHOT</version>
        <modelVersion>4.0.0</modelVersion>
        <packaging>pom</packaging>

        <parent>
                <groupId>com.myco</groupId>
                <artifactId>global-parent-pom</artifactId>
                <version>1.0-SNAPSHOT</version>
        </parent>

        <modules>
                <module>moduleA</module>
                <module>moduleB</module>
        </modules>

</project>
```
:sparkles: Important - notice that I am *not* explicitly specifying a groupId for this pom since it should inherit from the parent (com.myco) :sparkles: 

moduleA and moduleB use myProject's pom as their parent.  That's the setup - very standard.  

When I run `versions:set` at the myProject level I get the following NPE stacktrace:

```
[DEBUG] Applying change null:my-project-pom:1.0.0-SNAPSHOT -> 1.0.1-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO]
[INFO] moduleA ................................... FAILURE [  8.936 s]
[INFO] moduleB ......................................... SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 9.818 s
[INFO] Finished at: 2015-07-14T12:35:50-04:00
[INFO] Final Memory: 13M/156M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.codehaus.mojo:versions-maven-plugin:2.2:set (default-cli) on project droit-engine-pom: Execution default-cli of goal org.codehaus.mojo:versions-maven-plugin:2.2:set failed. NullPointerException -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.codehaus.mojo:versions-maven-plugin:2.2:set (default-cli) on project droit-engine-pom: Execution default-cli of goal org.codehaus.mojo:versions-maven-plugin:2.2:set failed.
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:224)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:116)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:80)
	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:51)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:120)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:347)
	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:154)
	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:582)
	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:214)
	at org.apache.maven.cli.MavenCli.main(MavenCli.java:158)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
Caused by: org.apache.maven.plugin.PluginExecutionException: Execution default-cli of goal org.codehaus.mojo:versions-maven-plugin:2.2:set failed.
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:143)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:208)
	... 19 more
Caused by: java.lang.NullPointerException
	at org.codehaus.mojo.versions.api.PomHelper.getModelEntry(PomHelper.java:1571)
	at org.codehaus.mojo.versions.SetMojo.applyChange(SetMojo.java:275)
	at org.codehaus.mojo.versions.SetMojo.execute(SetMojo.java:239)
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:132)
	... 20 more
```
Notice in the top line that the groupId is showing up as null.
### Workaround

If I explicitly add the groupId (`<groupId>com.myco</groupId>`) to the root level of myProject's pom, it works.

**Comment by tonit** [comment]
This looks still broke  with release 2.2 (from august 2015):
```
[ERROR] Failed to execute goal org.codehaus.mojo:versions-maven-plugin:2.2:set (default-cli) on project base: Execution default-cli of goal org.codehaus.mojo:versions-maven-plugin:2.2:set failed. NullPointerException -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.codehaus.mojo:versions-maven-plugin:2.2:set (default-cli) on project base: Execution default-cli of goal org.codehaus.mojo:versions-maven-plugin:2.2:set failed.
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:224)
```
Can anyone tell me want went wrong with the 2.2 timeline (bugfix in july, release in august)?
Any chance to get 2.3 out of the door? Just wondering ;)

**Comment by nicmon** [comment]
+1 - Still existing on Version 2.2

**Comment by csdcrussell** [comment]
Ran into this today on 2.2.

**Comment by carlobongiovanni** [comment]
It happened the same to me with 2.2 today.

**Comment by jxblum** [comment]
Still a problem with Maven version 3.3.9 and org.codehause.mojo:versions-maven-plugin:2.2.

````
$mvn --version
Apache Maven 3.3.9 (bb52d8502b132ec0a5a3f4c09453c07478323dc5; 2015-11-10T08:41:47-08:00)
Maven home: /Users/jblum/DEV/apache-maven-3.3.9
Java version: 1.8.0_66, vendor: Oracle Corporation
Java home: /Library/Java/JavaVirtualMachines/jdk1.8.0_66.jdk/Contents/Home/jre
Default locale: en_US, platform encoding: UTF-8
OS name: "mac os x", version: "10.11.2", arch: "x86_64", family: "mac"
````

**Comment by jschneider** [comment]
Still a problem. Run in it today..

**Comment by fripoli** [comment]
is there a roadmap for the release of 2.3 version with this fix?

**Comment by nishantmonu51** [comment]
Ran into it today, any release plans for fix ?

**Comment by nishantmonu51** [comment]
as a workaround - reverting the version to 2.1 worked for us. 
```mvn org.codehaus.mojo:versions-maven-plugin:2.1:set -DgenerateBackupPoms=false -DnewVersion=SOME_VERSION```

**Comment by sgandon** [comment]
thanks for the workaround, any idea on when this shall be fixed ?

**Comment by ryan-williams** [comment]
Seems like it's fixed in version 2.3
